File: redsheet-portal/src/app/database-models/v-lite-background-communication-mode.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vLiteBackgroundCommunicationMode {
    constructor(vLiteBackgroundCommunicationMode?: vLiteBackgroundCommunicationMode )  {
        let properties = [ "communicationModeId", "communicationModeName", "liteBackgroundCommunicationModeId", "liteProjectBackgroundId",];
        for (let property of properties)
            if (vLiteBackgroundCommunicationMode && vLiteBackgroundCommunicationMode[property])
                this[property] = vLiteBackgroundCommunicationMode[property];
    }
 
	communicationModeId : number =   0 ;
 
    @required()
    @maxLength(200)
	communicationModeName : string =   undefined;
 
    @range(1,2147483647)
	liteBackgroundCommunicationModeId : number =   undefined;
 
    @range(1,2147483647)
	liteProjectBackgroundId : number =   undefined;


}

File: RedSheet.DbEntities/Models/vLiteBackgroundCommunicationMode.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vLiteBackgroundCommunicationModes", Schema = "dbo")]
	[Serializable]
	public partial class vLiteBackgroundCommunicationMode  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int CommunicationModeId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string CommunicationModeName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int LiteBackgroundCommunicationModeId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int LiteProjectBackgroundId { get; set; } 

	public vLiteBackgroundCommunicationMode()
	{
    }

    }
}

